### ISSUE TITLE
Regarding text -> image retrieval

### ISSUE THREAD

**Issue by liumingzhu6060** [TIGER-AI-Lab/VLM2Vec#30]
My code is as follows:
import os
os.environ["CUDA_VISIBLE_DEVICES"] = "1"
from src.model import MMEBModel
from src.arguments import ModelArguments
from src.utils import load_processor
import torch
from transformers import HfArgumentParser, AutoProcessor
from PIL import Image
import numpy as np

model_args = ModelArguments(
    model_name='/home/liumingzhu/VLM2Vec/VLM2Vec-LLaVa-Next',
    pooling='last',
    normalize=True,
    model_backbone='llava_next')

processor = load_processor(model_args)

model = MMEBModel.load(model_args)
model.eval()
model = model.to('cuda', dtype=torch.bfloat16)

string = 'Find me an everyday image that matches the given caption: A woman wearing yellow clothes.'
inputs = processor(text=string,
                   images=None,
                   return_tensors="pt")
inputs = {key: value.to('cuda') for key, value in inputs.items()}
qry_output = model(qry=inputs)["qry_reps"]

inputs_1 = processor(text="<|image_1|> Represent the given image.",
                   images=Image.open('mydata/yellow_women_1.jpg'),
                   return_tensors="pt")
inputs_1 = {key: value.to('cuda') for key, value in inputs_1.items()}
tgt_output_1 = model(tgt=inputs_1)["tgt_reps"]

inputs_2 = processor(text="<|image_1|> Represent the given image.",
                   images=Image.open('mydata/yellow_women_2.jpg'),
                   return_tensors="pt")
inputs_2 = {key: value.to('cuda') for key, value in inputs_2.items()}
tgt_output_2 = model(tgt=inputs_2)["tgt_reps"]

inputs_3 = processor(text="<|image_1|> Represent the given image.",
                   images=Image.open('mydata/yellow_women_3.jpg'),
                   return_tensors="pt")
inputs_3 = {key: value.to('cuda') for key, value in inputs_3.items()}
tgt_output_3 = model(tgt=inputs_3)["tgt_reps"]

inputs_4 = processor(text="<|image_1|> Represent the given image.",
                   images=Image.open('mydata/yellow_women_4.jpg'),
                   return_tensors="pt")
inputs_4 = {key: value.to('cuda') for key, value in inputs_4.items()}
tgt_output_4 = model(tgt=inputs_4)["tgt_reps"]

inputs_5 = processor(text="<|image_1|> Represent the given image.",
                   images=Image.open('mydata/yellow_women_5.jpg'),
                   return_tensors="pt")
inputs_5 = {key: value.to('cuda') for key, value in inputs_5.items()}
tgt_output_5 = model(tgt=inputs_5)["tgt_reps"]

print(string, '=', model.compute_similarity(qry_output, tgt_output_1))
print(string, '=', model.compute_similarity(qry_output, tgt_output_2))
print(string, '=', model.compute_similarity(qry_output, tgt_output_3))
print(string, '=', model.compute_similarity(qry_output, tgt_output_4))

The result is as follows, different images but same similarity:
![image](https://github.com/user-attachments/assets/611679e4-9d52-4584-8a5a-6036f810d642)

Excuse me, is there a problem with my code？

**Comment by liumingzhu6060** [comment]
another question：
Use official code，retrieve the caption "A woman wearing yellow clothes",results as follows:
![image](https://github.com/user-attachments/assets/9ed3b865-5f71-4fae-a8b6-14d56805f437)
The results show that the model cannot solve the problem caused by background color

**Comment by liumingzhu6060** [comment]
another question：
Use official code，retrieve the caption "Ride an electric bike without wearing a helmet",results as follows:
![image](https://github.com/user-attachments/assets/83823ca0-551e-451f-bea4-21d7f6199f53)
The results show that the model cannot understand the meaning of 'without'.

**Comment by XMHZZ2018** [comment]
@liumingzhu6060 

Thank you for your interest in our work! It seems that you are using an incorrect image token placeholder. Specifically, you are using Phi-3-v's image placeholder while working with the LLAVA-Next version of VLM2Vec. As a result, the image isn't being processed, which is why you're getting the same result. For LLAVA-Next VLM2Vec, the correct image token is <image> rather than <|image_1|>. Please refer to our demo code here: https://github.com/TIGER-AI-Lab/VLM2Vec/blob/main/scripts/llava_next/demo.py

**Comment by liumingzhu6060** [comment]
Thank you for your reply. The result is normal now, but it seems that there are two issues with the model.
1、The model cannot understand  the word "without" or "not".
2、The model's ability for compositional reasoning(attributes + objects),  the background color can affect retrieval results.
Here are two examples：

![Image](https://github.com/user-attachments/assets/6b8fb5ab-77ee-4875-95ca-502d685a601e)

![Image](https://github.com/user-attachments/assets/e6472aca-1186-4619-856a-154422526633)

![Image](https://github.com/user-attachments/assets/d3cc21bb-39e0-42f3-898a-14420de4bd94)

![Image](https://github.com/user-attachments/assets/c02fa563-96c0-4c6f-9640-a74a505c93df)

Do you have any ideas on how to solve these two problems?

**Comment by liumingzhu6060** [comment]
another case：

![Image](https://github.com/user-attachments/assets/b4d24d91-8dbb-4cec-80a4-7ff69c8ff201)

![Image](https://github.com/user-attachments/assets/ae461fd1-bfba-4b06-8158-04ad2e73000c)

![Image](https://github.com/user-attachments/assets/147e2fb3-b8a3-4098-8aca-a8cf97ea33d5)

It seems that the model does not understand semantic fusion and only uses individual words for matching.

**Comment by XMHZZ2018** [comment]
@liumingzhu6060 

Thank you for sharing these very interesting cases! We do use some training data that requires strong reasoning rather than just simple words. However, based on the cases you shared, there is still room for improvement. I believe the most direct way to address these issues is by incorporating hard negatives during training. We are currently working on the next version of VLM2Vec, which includes hard negatives in the training process. Hopefully, this will resolve the issues you mentioned. I’ll keep you updated once we’ve completed the training.
